synapse/http/server.py

```diff
@@ -706,15 +707,35 @@ def respond_with_json_bytes(
     if send_cors:
         set_cors_headers(request)
 
-    # note that this is zero-copy (the bytesio shares a copy-on-write buffer with
-    # the original `bytes`).
-    bytes_io = BytesIO(json_bytes)
-
-    producer = NoRangeStaticProducer(request, bytes_io)
-    producer.start()
+    _write_json_bytes_to_request(request, json_bytes)
     return NOT_DONE_YET
 
 
+def _write_json_bytes_to_request(request: Request, json_bytes: bytes) -> None:
```

Comment:
From https://github.com/matrix-org/synapse/pull/10844#discussion_r713034847

> is this specific to being json bytes, or could it be used for any byte sequence?

It can be anything that slices to `bytes`, but I don't see any in built typing for that (`ByteString` seems to just be an alias for `Sequence[int]]`...). I can add a protocol, or do `Union[bytes, bytearray]`? I'm not sure its worth its worth it though

Reply:
I'm mostly asking: why don't we just call it `write_bytes_to_request` (and also `s/json_bytes/bytes_to_write/` or something)?

Reply:
Ooooooooooooooooooooooooooooooooooooooooooooh.

Yup, happy to rename.